pulsar-rpc-contrib/src/test/java/org/apache/pulsar/rpc/contrib/SimpleRpcCallTest.java

```diff
@@ -0,0 +1,476 @@
+    private final String requestSubBase = requestTopic + "-sub";
+    private final String replySubBase = replyTopic + "-sub";
+    private final Duration replyTimeout = Duration.ofSeconds(3);
+    private final String synchronousMessage = "SynchronousRequest";
+    private final String asynchronousMessage = "AsynchronousRequest";
+    private final Schema<TestRequest> requestSchema = Schema.JSON(TestRequest.class);
+    private final Schema<TestReply> replySchema = Schema.JSON(TestReply.class);
+
+    private PulsarAdmin pulsarAdmin;
+    private PulsarClient pulsarClient;
+    private PulsarRpcClient<TestRequest, TestReply> rpcClient;
+    private PulsarRpcServer<TestRequest, TestReply> rpcServer;
+
+    private Function<TestRequest, CompletableFuture<TestReply>> requestFunction;
+    private BiConsumer<String, TestRequest> rollbackFunction;
+
+    @BeforeMethod(alwaysRun = true)
+    protected void setup() throws Exception {
+        pulsarAdmin = PulsarAdmin.builder().serviceHttpUrl("http://localhost:8080").build();
+        pulsarAdmin.topics().createPartitionedTopic(requestTopic, 10);
+        pulsarAdmin.topics().createPartitionedTopic(replyTopic, 10);
+        pulsarClient = PulsarClient.builder().serviceUrl("pulsar://localhost:6650").build();
+    }
+
+    @AfterMethod(alwaysRun = true)
+    protected void cleanup() throws Exception {
+        if (rpcServer != null) {
+            rpcServer.close();
+        }
+        rpcClient.close();
+        pulsarClient.close();
+        pulsarAdmin.topics().deletePartitionedTopic(requestTopic);
+        pulsarAdmin.topics().deletePartitionedTopic(replyTopic);
+        pulsarAdmin.close();
+    }
+
+    @Test
+    public void testRpcCall() throws Exception {
+        ProducerBuilder<TestRequest> requestProducerBuilder = pulsarClient.newProducer(requestSchema)
+                .topic(requestTopic)
+                .enableBatching(false)
+                .producerName("requestProducer");
+
+        // 1.Create PulsarRpcClient
+        rpcClient = createPulsarRpcClient(pulsarClient, requestProducerBuilder, null, null);
+
+        // 2.Create PulsarRpcServer
+        final int defaultEpoch = 1;
+        AtomicInteger epoch = new AtomicInteger(defaultEpoch);
+        // What do we do when we receive the request message
+        requestFunction = request -> {
+            epoch.getAndIncrement();
+            return CompletableFuture.completedFuture(new TestReply(request.value() + "-----------done"));
+        };
+        // If the server side is stateful, an error occurs after the server side executes 3-1, and a mechanism for
+        // checking and rolling back needs to be provided.
+        rollbackFunction = (id, request) -> {
+            if (epoch.get() != defaultEpoch) {
+                epoch.set(defaultEpoch);
+            }
+        };
+        rpcServer = createPulsarRpcServer(pulsarClient, requestSubBase, requestFunction, rollbackFunction, null);
+
```

Comment:
Question6: I think writing just one testcase is not enough. do you want to add a best practice? you currently divide it into three modules: client/common/server. If I stand in the user's perspective, how should I use your feature? For example:

1. what imports does the client make, and what does the access demo look like?
2. what imports does the broker make, and what is the startup method like?

Reply:
Makes sense, I will update more test cases later.

Reply:
@StevenLuMT The following comment may refactor the code and may have different usage and exception handling. This test case extension needs to be completed after confirmation.